server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -509,8 +515,17 @@ export class ChatDatabase {
                     this.#features.logging.error(`Error counting toolResults: ${String(e)}`)
                 }
             }
+            // Count characters in tool spec for the current user message
+            if (message.userInputMessageContext?.tools) {
+                try {
+                    for (const toolSpec of message.userInputMessageContext.tools) {
+                        count += JSON.stringify(toolSpec).length
+                    }
+                } catch (e) {
+                    this.#features.logging.error(`Error counting tool spec length: ${String(e)}`)
```

Comment:
This will log, and then silently fail, is that ok?

Reply:
Yeah that's ok for this. I don't think there is much we can do in that event. We should still try to continue with the request/job